src/main/java/org/disq_bio/disq/impl/formats/bam/HeaderlessBamOutputFormat.java

```diff
@@ -31,7 +32,7 @@
 
     public BamRecordWriter(Configuration conf, Path file, SAMFileHeader header) throws IOException {
       this.out = file.getFileSystem(conf).create(file);
-      BlockCompressedOutputStream compressedOut = new BlockCompressedOutputStream(out, null);
+      BlockCompressedOutputStream compressedOut = new BlockCompressedOutputStream(out, (File) null);
```

Comment:
I hate htsjdk.

Anyone object to making an assertion in this library that all method parameters should be nonnull by default?
```java
@ParametersAreNonnullByDefault
package foo;

import javax.annotation.ParametersAreNonnullByDefault;
```

https://github.com/google/guava/wiki/UsingAndAvoidingNullExplained#using-and-avoiding-null

Reply:
@heuermh I'm not very familiar with the various nullness annotations. What does adding that do?  Is it just a statement of principle or is enforced somehow?

Reply:
As far as I know, a statement of principle, although there may be tools that use that annotation in static code analysis.

Throughout the guava library method parameters are checked via [`Preconditions`](https://github.com/google/guava/wiki/PreconditionsExplained) to fail fast on nulls.  Adding a dependency on guava comes with its own set of problems though.

I'd be happy to simply say we're not going to design our API to accept nulls as method parameters.

Reply:
+1 to not allowing nulls in the API. I'd rather not have a Guava dependency though.